File: LLM-Translations/GPT-4/codenet/Go/Java/s865387202.java

import java.util.*;

class City implements Comparable<City> {
    int prefecture;
    int position;
    int year;

    public City(int prefecture, int year, int position) {
        this.prefecture = prefecture;
        this.year = year;
        this.position = position;
    }

    @Override
    public int compareTo(City other) {
        return Integer.compare(this.year, other.year);
    }
}

public class s865387202 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();
        HashMap<Integer, ArrayList<City>> h = new HashMap<>();

        for (int i = 0; i < m; i++) {
            int x = sc.nextInt();
            int y = sc.nextInt();
            h.putIfAbsent(x, new ArrayList<>());
            h.get(x).add(new City(x, y, i));
        }

        String[] results = new String[m];

        for (ArrayList<City> cities : h.values()) {
            Collections.sort(cities);
            for (int idx = 0; idx < cities.size(); idx++) {
                City ci = cities.get(idx);
                results[ci.position] = String.format("%06d%06d", ci.prefecture, idx + 1);
            }
        }
        System.out.println(String.join("\n", results));
    }
}

File: LLM-Translations/GPT-4/codenet/Go/C/s865387202.c

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

typedef struct {
    int prefecture;
    int position;
    int year;
} city;

int compare(const void *a, const void *b) {
    city *cityA = (city *)a;
    city *cityB = (city *)b;
    return cityA->year - cityB->year;
}

int main() {
    int n, m;
    scanf("%d %d", &n, &m);
    city cities[m];

    for (int i = 0; i < m; i++) {
        int x, y;
        scanf("%d %d", &x, &y);
        cities[i].prefecture = x;
        cities[i].year = y;
        cities[i].position = i;
    }

    qsort(cities, m, sizeof(city), compare);

    char results[m][13];
    int count[n + 1];
    memset(count, 0, sizeof(count));

    for (int i = 0; i < m; i++) {
        int prefecture = cities[i].prefecture;
        int idx = ++count[prefecture];
        snprintf(results[cities[i].position], 13, "%06d%06d", prefecture, idx);
    }

    for (int i = 0; i < m; i++) {
        printf("%s\n", results[i]);
    }

    return 0;
}

// 
